### PR TITLE
Adding scarpet hooks to events

### PR DESCRIPTION
**Pull request by Ghoulboy78**
Idk abt the Chinese bit.

Suggestion by @Firigion (Could I have ur feedback as well?)
https://discord.com/channels/645464470633840651/645484896529416235/804390837445197844

### CODE REVIEW

**Comment by Ghoulboy78** [comment]
wait, it's still broken rn.

**Comment by Ghoulboy78** [comment]
k, now it's working

File: README.md [hunk 1]
```diff
@@ -778,6 +786,36 @@ Show the detail statistic of specific entity type. You can specify which part of
...
+
+## Functions
+
+### `register_block(pos)`
+
+Registers a block position to be tracked with events by scarpet. It is added to a global tracker (which you can view with
+`registered()`). Blocks in this list will trigger events in `__on_block_event` if they occur, independently of the loggers.
+Returns `true` if block was not previously in list, and `false` if it was.
+
+### `deregister_block(pos)`
```
Comment (Fallen-Breath) [inline README.md]: does `unregister` sound better than `deregister`? never heard of `deregister` before

File: README.md [hunk 2]
```diff
@@ -778,6 +786,36 @@ Show the detail statistic of specific entity type. You can specify which part of
...
+
+Removes a block position from the list of block positions to be tracked. Returns `true` if block was previously in the
+list, and `false` if not.
+
+### `registered()`, `registered(pos)`
```
Comment (Fallen-Breath) [inline README.md]: I will suggest to split this 2 funtionalities into 2 different functions, that will make things cleaner and less confused

File: README.md [hunk 3]
```diff
@@ -778,6 +786,36 @@ Show the detail statistic of specific entity type. You can specify which part of
...
+
+Without any arguments, returns the list of block positions which are tracked by scarpet to trigger the block event events
+within scarpet. With a position, will return `true` or `false` based on whether or not that position is in the list.
+
+## Events
+
+### `__on_block_event(type, pos)`
```
Comment (Fallen-Breath) [inline README.md]: idk if there's a better name than `block_event` since it might confused with the piston block event thing

**Comment by Fallen-Breath** [comment]
Is it intentional in the logic that it does not distinguish which world/dimension the event is in

**Comment by Ghoulboy78** [comment]
oh wait ye I forgot im an idiot

File: README.md (hunk 3 above)
Comment (Ghoulboy78) [reply README.md]: ye, i wasn't sue what to add instead...

Comment (Ghoulboy78) [reply README.md]: cos in the code it's BaseEvent, but `__on_base_event` looks bad...

File: README.md (hunk 1 above)
Comment (Ghoulboy78) [reply README.md]: k

File: README.md (hunk 3 above)
Comment (Fallen-Breath) [reply README.md]: u can give it another name like `microtiming_event`

Comment (Ghoulboy78) [reply README.md]: ye, great idea!

**Comment by Ghoulboy78** [comment]
k added it. Now idk abt the chinese translation for the README cos I don't speak chinese...

**Comment by Fallen-Breath** [comment]
> Now idk abt the chinese translation for the README cos I don't speak chinese...

I will handle the chinese part

File: src/main/java/carpettisaddition/script/BlockEvents.java [hunk 4]
```diff
@@ -27,27 +27,27 @@ public BlockEvents(String name, int reqArgs, boolean isGlobalOnly) {
...
         if(event instanceof ScheduleTileTickEvent)
-            BLOCK_EVENT.onBlockEvent("scheduled_tile_tick", pos);
+            MICRO_TIMING_EVENT.onBlockEvent("scheduled_tile_tick", pos);
     }
 
 
     public void onBlockEvent(String type, BlockPos pos){}
 
-    public static BlockEvents BLOCK_EVENT = new BlockEvents("block_event", 2, true) {
+    public static BlockEvents MICRO_TIMING_EVENT = new BlockEvents("microtiming_event", 2, true) {
```
Comment (Fallen-Breath) [inline src/main/java/carpettisaddition/script/BlockEvents.java]: dont forget the class name xd

Comment (Ghoulboy78) [reply src/main/java/carpettisaddition/script/BlockEvents.java]: oh ye lol

File: README.md [hunk 5]
```diff
@@ -793,25 +794,28 @@ For example, `/lifetime creeper` shows all statistic of creeper in detail, and `
...
 list, and `false` if not.
 
-### `registered()`, `registered(pos)`
+### `registered_blocks()`
 
-Without any arguments, returns the list of block positions which are tracked by scarpet to trigger the block event events
-within scarpet. With a position, will return `true` or `false` based on whether or not that position is in the list.
+Returns the list of block positions which are tracked by scarpet to trigger the block event events within scarpet.
+
+### `registered(pos)`
```
Comment (Fallen-Breath) [inline README.md]: `is_registered(pos)` would be nicer imo

Comment (Ghoulboy78) [reply README.md]: k

File: README.md [hunk 6]
```diff
@@ -705,34 +720,6 @@ List information of all current raids
...
-This tracker is mostly used to debug mobfarms. It aims to track the process from mob starting affecting the mobcap to mob being removed from the mobcap. The spawning tracking part of it doesn't cover every kind of mob spawning reasons
-
-Other than being removed from the world, if a mob becomes persistent for the first time like nametagged or item pickup, it will be marked as removal too. If a mob doesn't count towards the mobcap when it spawns, it will not be tracked
-
-This tracker also tracks lifetime of items and xp orbs from mob and block drops as an additional functionality. Note that it doesn't track all item / xp orb spawning, so you'd better have a test before actually using it
-
-Adding a `realtime` suffix to the command will turn the rate result from in-game time based to realtime based
-
-### tracking
-
```
Comment (Fallen-Breath) [inline README.md]: looks like you have messed up the readme structure. plz check the orignial readme and fix it

**Review by Fallen-Breath — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -703,13 +718,6 @@ Attributes:
 
 List information of all current raids
 
-### tracking
-
-`/raid tracking [<start|stop|restart|realtime>]`
-
-Start a raid tracking to gather statistics from ongoing raids
-
-
```
Comment (Fallen-Breath) [inline README.md]: and this `/raid tracking` thing

**Review by Fallen-Breath — APPROVED** [review]

**Comment by Fallen-Breath** [comment]
I added a rule `microTiming` check before looking up in the hashset `trackedPositions` in `, so it doesnt take too much performance when u dont need it

**Pull request merged**